docs/building-images.md

```diff
@@ -157,7 +157,46 @@ operatingSystem:
   * `additionalRepos` - Optional; List of third-party RPM repositories that will be added to the package manager of the OS.
   * `sccRegistrationCode` - Optional; SUSE Customer Center registration code, used to connect to SUSE's internal RPM repositories.
 
-## SUSE Manager (SUMA)
+### Kubernetes
+
+The Kubernetes configuration section is another entirely optional one.
+It contains all necessary settings to configure and bootstrap a Kubernetes cluster.
+
+```yaml
+kubernetes:
+  version: v1.29.0+rke2r1
+  network:
+    apiVIP: 192.168.122.100
+    apiHost: api.cluster01.hosted.on.edge.suse.com
+  nodes:
+    - hostname: node1.suse.com
+      type: server
+    - hostname: node2.suse.com
+      type: server
+      initializer: true
+    - hostname: node3.suse.com
+      type: agent
+    - hostname: node4.suse.com
+      type: server
+    - hostname: node5.suse.com
+      type: agent
+  manifests:
+    urls:
+      - https://k8s.io/examples/application/nginx-app.yaml
+```
+
+* `version` - Required; version string of a particular k3s or RKE2 release e.g.`v1.28.7+k3s1` or `v1.29.0+rke2r1`
+* `network` - Required for multi node clusters, optional for single node clusters; network configuration for bootstrapping a cluster
+  * `apiVIP` - Required for multi node clusters, optional for single node clusters; IP address which will serve as the cluster LoadBalancer (backed by MetalLB)
+  * `apiHost` - Optional; domain address for accessing the cluster
+* `nodes` - Required for multi node clusters; list of all nodes forming the cluster
+  * `hostname` - Required; hostname (usually FQDN) which identifies the particular node
+  * `type` - Required; Kubernetes node type - either `server` (for control plane nodes) or `agent` (for worker nodes)
+  * `initializer` - Optional; identifier of the node forming the cluster. If unset, the first server in the node list will be selected as the initializer.
+* `manifests` - Optional; manifests to apply to the cluster
```

Comment:
Are these applied _after_ the cluster is up or are these applied as extra manifests https://docs.rke2.io/advanced#auto-deploying-manifests ?

Reply:
The difference is important because if you _need_ those manifests to be up when the cluster is ready (like metallb) you cannot apply them before the cluster is up... and viceversa, maybe you need the cluster to be fully up for the workload to be deployed.